### ISSUE TITLE
How to pause the recording?

### ISSUE THREAD

**Issue by FazliMolaJan** [HBiSoft/HBRecorder#5]
Like I want to pause the screen recording while it is recording and restart it again from there? How I can do it?

**Comment by HBiSoft** [comment]
I will implement this soon.

**Comment by HBiSoft** [comment]
The only way I was able to do this is to create different video files each time the video is paused/resumed and then merge them together.

There is a library called [Mp4Parser](https://github.com/sannies/mp4parser) that can merge the files.

Implement the library into your project then you can merge the files like this:

```java
public static boolean mergeMediaFiles(boolean isAudio, String sourceFiles[], String targetFile) {
    try {
        String mediaKey = isAudio ? "soun" : "vide";
        List<Movie> listMovies = new ArrayList<>();
        for (String filename : sourceFiles) {
            listMovies.add(MovieCreator.build(filename));
        }
        List<Track> listTracks = new LinkedList<>();
        for (Movie movie : listMovies) {
            for (Track track : movie.getTracks()) {
                if (track.getHandler().equals(mediaKey)) {
                    listTracks.add(track);
                }
            }
        }
        Movie outputMovie = new Movie();
        if (!listTracks.isEmpty()) {
            outputMovie.addTrack(new AppendTrack(listTracks.toArray(new Track[listTracks.size()])));
        }
        Container container = new DefaultMp4Builder().build(outputMovie);
        FileChannel fileChannel = new RandomAccessFile(String.format(targetFile), "rw").getChannel();
        container.writeContainer(fileChannel);
        fileChannel.close();
        return true;
    }
    catch (IOException e) {
        Log.e(LOG_TAG, "Error merging media files. exception: "+e.getMessage());
        return false;
    }
}
```

You can also use FFmpeg to merge the files.

---
I will be closing this issue because this can be implemented by the developer instead of adding it to the library.

**Comment by chiragthummar** [comment]
@HBiSoft  i have tried to merge video using above code but not able to do because mp4parser lib gives error.
will you help me in merging video with example.

**Comment by HBiSoft** [comment]
@chiragthummar 
I just did a test and it is working perfectly.
What error are you getting?

**Comment by chiragthummar** [comment]
its give me error like below


```
W/System.err: java.io.FileNotFoundException: /storage/emulated/0/HD2020-08-31-18-55-29.mp4 (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:231)
        at java.io.FileInputStream.<init>(FileInputStream.java:165)
        at com.googlecode.mp4parser.FileDataSourceImpl.<init>(FileDataSourceImpl.java:20)
W/System.err:     at com.googlecode.mp4parser.authoring.container.mp4.MovieCreator.build(MovieCreator.java:38)
        at com.qp.mp4parserdemo.Mp4ParserUtils.mergeVideo(Mp4ParserUtils.java:36)
        at com.qp.mp4parserdemo.MainActivity$1.onClick(MainActivity.java:37)
        at android.view.View.performClick(View.java:6733)
        at android.view.View.performClickInternal(View.java:6691)
        at android.view.View.access$3400(View.java:802)
        at android.view.View$PerformClick.run(View.java:26499)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:226)
W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:7178)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942)
```

**Comment by chiragthummar** [comment]
and sometime it give this error as well.

W/System.err: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List com.coremedia.iso.boxes.MovieBox.getBoxes(java.lang.Class)' on a null object reference
W/System.err:     at com.googlecode.mp4parser.authoring.container.mp4.MovieCreator.build(MovieCreator.java:51)
        at com.googlecode.mp4parser.authoring.container.mp4.MovieCreator.build(MovieCreator.java:38)
        at com.hbisoft.hbrecorderexample.Mp4ParserUtils.mergeVideo(Mp4ParserUtils.java:34)
        at com.hbisoft.hbrecorderexample.MainActivity$2.onClick(MainActivity.java:223)
        at android.view.View.performClick(View.java:6733)
        at android.view.View.performClickInternal(View.java:6691)
        at android.view.View.access$3400(View.java:802)
        at android.view.View$PerformClick.run(View.java:26499)
        at android.os.Handler.handleCallback(Handler.java:873)
W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:226)
        at android.app.ActivityThread.main(ActivityThread.java:7178)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942)

**Comment by chiragthummar** [comment]
will you please give me full example of merging video ?

**Comment by HBiSoft** [comment]
@chiragthummar 
I will provide an example within the next 2-3 hours.

**Comment by chiragthummar** [comment]
thanks

**Comment by chiragthummar** [comment]
Hello
does below code work in android Q or not ?

```
if (mMediaRecorder != null) {
            mMediaRecorder.pause();
        }
```

**Comment by HBiSoft** [comment]
@chiragthummar 
It is now available in `0.1.13`.

You can pause the recording by calling:
```java
hbRecording.pauseScreenRecording();

```
You can resume the recording by calling:
```java
hbRecording.resumeScreenRecording();
```

This is only available for devices running 24> and it is important to check if the recording has started or was paused using `hbRecorder.isBusyRecording();` and `hbrecorder.isRecordingPaused();`.

Hope you find this helpful.

**Comment by chiragthummar** [comment]
Thank you so much for your quick answer.
